index.html

```diff
@@ -1787,6 +1951,310 @@ <h2>
+        URL in the scope of the application that can handle launches with
+        [=file types=] it accepts. It has the following members:
+      </p>
+      <ul>
+        <li>[=file handler/action=]
+        </li>
+        <li>[=file handler/name=]
+        </li>
+        <li>[=file handler/accept=]
+        </li>
+        <li>[=file handler/icons=]
+        </li>
+      </ul>
+      <p>
+        A user agent can use these members to associate the web application
+        with [=file type=] on the operating system.
+      </p>
+      <p>
+        A <dfn>file type</dfn> can be defined by a [=mime type=] and/or [=file
+        extension=]. A file belongs to a file type if the OS determines it to
+        have a mime type and/or its name ends with a certain [=file
+        extension=]. A <dfn>file extension</dfn> is a string that start with a
+        "." and only contains [=valid suffix code points=]. A <dfn>valid suffix
+        code point</dfn> is a [=code point=] that is [=ASCII alphanumeric=],
+        U+002B (+), or U+002E (.). Additionally, [=file extensions=] are are
+        limited to a length of 16 code points.
+      </p>
+      <aside class="note">
+        <p>
+          Note: These code points were chosen to support most pre-existing file
+          formats. The vast majority of file extensions are purely
+          alphanumeric, but compound extensions (such as .tar.gz) and
+          extensions such as .c++ for C++ source code are also fairly common,
+          hence the inclusion of + and . as allowed code points.
+        </p>
+      </aside>
+      <aside class="note">
+        <p>
+          The selection of an app to handle a file launch if there are multiple
+          apps registerd for the file's [=file type=] is functionality left to
+          the operating system.
+        </p>
+      </aside>
+      <section>
+        <h3>
+          `action` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
+          "file handler">action</dfn></code> member is a <a>string</a> that
+          represents a relative URL of the [=manifest/start_url=] origin that
+          is [=manifest/within scope=] of a [=Document/processed manifest=].
+          This URL will be navigated to in the steps to [=execute a file
+          handler launch=].
+        </p>
+      </section>
+      <section>
+        <h3>
+          `name` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
+          "file handler">name</dfn></code> member is a <a>string</a> that
+          represents a name to be provided to the user's operating system for
+          use with "Open with..." UX flows. This MAY be replaced by the user
+          agent for privacy & security reasons.
+        </p>
+      </section>
+      <section data-cite="MIMETYPE">
+        <h3>
+          `accept` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
```

Comment:
Yes. This is why the file extensions are required. I've attempted to clarify the wording here.

The goal is greater consistency of user experience across OSes and user agents.

Because certain operating systems require mime types for the association (linux via xdg-mime), we enforce that site authors must provide mime types. Any attempt by the user agent to smartly map from extension to mime type or vice versa would cause different user agents to behave differently since the mappings would not be identical.

However, it's also enforced that the file extension matches what's provided in the manifest. This may happen at the OS level or at the user agent level. For example, `"text/*": [".txt"]` would not match a `.csv` file anywhere, even on Linux where the file is recognized to match that MIME type. This would cause different OSes to behave differently.